analytics.js

```diff
@@ -1023,6 +1023,11 @@ class Analytics {
       }
     }
 
+    if (options && options.useBeacon === true) {
+      this.eventRepository.useBeacon = options.useBeacon;
+      this.eventRepository.initializeTransportMechanism();
```

Comment:
This should be internally handled by the event repository when it is triggered to start queuing. SDK should not be aware of these details.

Reply:
done